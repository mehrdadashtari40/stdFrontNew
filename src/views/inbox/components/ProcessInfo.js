import React, {Component} from 'react';
import jMoment from 'moment-jalaali';
import {withTranslation} from 'react-i18next';
import {compose} from 'redux';

function ProcessInfo(props) {
    return (
        <div>
            <table className="table table-bordered">
                <tbody>
                <tr>
                    <td> {props.t('Title')}</td>
                    <td> {props.processInfo.PRO_TITLE ? props.processInfo.PRO_TITLE : props.t('Not Value')} </td>
                </tr>
                <tr>
                    <td> {props.t('Description')}</td>
                    <td>{props.processInfo.PRO_DESCRIPTION ? props.processInfo.PRO_DESCRIPTION : props.t('Not Value')}</td>
                </tr>
                <tr>
                    <td> {props.t('Category')}</td>
                    <td>{props.processInfo.PRO_CATEGORY_LABEL}</td>
                </tr>
                <tr>
                    <td> {props.t('Author')}</td>
                    <td>{props.processInfo.PRO_AUTHOR}</td>
                </tr>
                <tr>
                    <td>{props.t('Create Date')}</td>
                    <td>{jMoment(props.processInfo.PRO_CREATE_DATE).format('HH:mm:ss jYYYY/jMM/jDD')}</td>
                </tr>
                </tbody>
            </table>
        </div>
    );
}
export default withTranslation()(ProcessInfo);