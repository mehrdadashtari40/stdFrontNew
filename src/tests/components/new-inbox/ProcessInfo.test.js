import React from 'react';
import { shallow } from 'enzyme';
import ProcessInfo from '../../../views/inbox/components/ProcessInfo';
jest.mock('react-i18next', () => ({
	withTranslation: () => Component => {
		Component.defaultProps = { ...Component.defaultProps, t: () => '' };
		return Component;
	},
}));
test('Make Snapshot file , ProcessInfo Component', () => {
	let fakeProcessInfo = {
		PRO_UID: '1271994115e57596c953e45081760466',
		PRO_ID: 77,
		PRO_TITLE: 'test-unassigned',
		PRO_DESCRIPTION: 'test unassigned',
		PRO_PARENT: '1271994115e57596c953e45081760466',
		PRO_TIME: 1,
		PRO_TIMEUNIT: 'DAYS',
		PRO_STATUS: 'ACTIVE',
		PRO_TYPE_DAY: '',
		PRO_TYPE: 'NORMAL',
		PRO_ASSIGNMENT: 'FALSE',
		PRO_SHOW_MAP: 0,
		PRO_SHOW_MESSAGE: 0,
		PRO_SUBPROCESS: 0,
		PRO_TRI_CREATE: '',
		PRO_TRI_OPEN: '',
		PRO_TRI_DELETED: '',
		PRO_TRI_CANCELED: '',
		PRO_TRI_PAUSED: '',
		PRO_TRI_REASSIGNED: '',
		PRO_TRI_UNPAUSED: '',
		PRO_TYPE_PROCESS: 'PUBLIC',
		PRO_SHOW_DELEGATE: 0,
		PRO_SHOW_DYNAFORM: 0,
		PRO_CATEGORY: '9111058575ce9344879c237056865651',
		PRO_SUB_CATEGORY: '',
		PRO_INDUSTRY: 0,
		PRO_UPDATE_DATE: '2020-02-27 09:27:11',
		PRO_CREATE_DATE: '2020-02-27 09:23:48',
		PRO_CREATE_USER: '00000000000000000000000000000001',
		PRO_HEIGHT: 5000,
		PRO_WIDTH: 10000,
		PRO_TITLE_X: 0,
		PRO_TITLE_Y: 0,
		PRO_DEBUG: 0,
		PRO_DYNAFORMS: false,
		PRO_DERIVATION_SCREEN_TPL: '',
		PRO_COST: 0,
		PRO_UNIT_COST: '',
		PRO_ITEE: 1,
		PRO_CATEGORY_LABEL: 'شرکت آرین',
		PRO_BPMN: 1,
		PRO_AUTHOR: 'Administrator admin',
	};

	const wrapper = shallow(<ProcessInfo processInfo={fakeProcessInfo} />);
	expect(wrapper).toMatchSnapshot();
});
