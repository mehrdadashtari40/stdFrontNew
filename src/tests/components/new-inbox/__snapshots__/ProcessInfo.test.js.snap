// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Make Snapshot file , ProcessInfo Component 1`] = `
<div>
  <table
    className="table table-bordered"
  >
    <tbody>
      <tr>
        <td>
           
        </td>
        <td>
           
          test-unassigned
           
        </td>
      </tr>
      <tr>
        <td>
           
        </td>
        <td>
          test unassigned
        </td>
      </tr>
      <tr>
        <td>
           
        </td>
        <td>
          شرکت آرین
        </td>
      </tr>
      <tr>
        <td>
           
        </td>
        <td>
          Administrator admin
        </td>
      </tr>
      <tr>
        <td />
        <td>
          09:23:48 1398/12/08
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
